frappe/core/doctype/user/user.js

```diff
@@ -341,6 +341,13 @@ frappe.ui.form.on("User", {
 			frappe.ui.toolbar.clear_cache();
 		}
 	},
+	user_type: function(frm) {
+		if(frm.doc.user_type == "Website User"){
+			frm.set_value("role_profile_name", "");
+			frm.set_value("roles", []);
+			frm.roles_editor.reset();
```

Comment:
This isn't correct. There can be non-desk roles which should be preserved.

Reply:
That means we have to remove the roles with desk access only.